### ISSUE TITLE
0.8.0.7 - Entering World takes forever, unable to join server, server-restart as workaround

### ISSUE THREAD

**Issue by Rycoviac** [StrangeLoopGames/EcoIssues#11684]
This Bug is similar to #11534 but this time, most players are completly unable to join the server and have a forever spinning cog.
Only a server-restart after a ~hour runtimes enables the players to join the server.
No error in logs or server UI, no updates on CPU-Time or in/out data on this players while joining but their characters spawn/added in the world, while the client of these players stuck on loading.

We have client logs from two players for this but could related to other bugs as well:
https://pastebin.com/34EyNtVA <<<kicked after cutting tree, unable to join after that (could be unrelated, who knows)

https://pastebin.com/F9ja4ejv
https://pastebin.com/HueTMB2Q

**Comment by Praefator** [comment]
Relevant loglines in https://pastebin.com/F9ja4ejv are at 1046 and onwards

**Comment by mirasrael** [comment]
From log it seems related to: https://github.com/StrangeLoopGames/EcoIssues/issues/11685

**Comment by Rycoviac** [comment]
> From log it seems related to: #11685

yea, the log: https://pastebin.com/34EyNtVA is related to this, but this player is also now unable to join. Sorry if this is a bit confusing. Its not that easy to differentiate all bugs anymore.

**Comment by mirasrael** [comment]
Probably server become to invalid state after this error happens. I'll build prepare custom server build for you for testing.

**Comment by mirasrael** [comment]
Can you try with exe from another issue (https://github.com/StrangeLoopGames/EcoIssues/issues/11685) and let me know if it helps?

**Comment by Rycoviac** [comment]
@mirasrael We will try it asap and give you feedback on this. Since it don't happen right after the restart, we have to wait and observe.

**Comment by mirasrael** [comment]
@Rycoviac Also if it happen again can you please make process dump and attach to this issue? you can do it from Task Manager, right click on Eco Server process, then Create Dump File

**Comment by Praefator** [comment]
Okay, I'll do so. He posted on my behalf but I'm the admin of said server. I estimate about 3+ hours before I can test, as I will need to get home from work first and have the server run for a while

**Comment by Rycoviac** [comment]
@mirasrael So far, we havn't had that problem again. Player was able to connect the whole time.

Also we can see also improvement, that we don't get disconnects anymore while server seems to have very large latencys. Until the custom build, if we encounterd heavy laggs, often all the players was disconnected. This seems to be gone, too.

Speaking of large latency: It seems that over time, maybe again after 1-2h runtime, the server gets very "slow" without(!) unusual high CPU or RAM or CPUtime/Player. Large delays of 1-5 seconds for dropping items in Inventorys, and shift+move items from one in another inventory CAN cause an lag up to ~20 seconds with red sign. This is serverwide. Also delays with entering vehicles and opening trunks/inventorys so player think the server didn't recognise it and press E more often.
Our server is open for several weeks and updates, so this could be something else. Our bloated DB or something. But I thought it would be good to mention this.

tl;dr: No more forever spinning cog for players. Specific problem solved with the custom build.

**Comment by mirasrael** [comment]
@Rycoviac thanks for feedback. We focused on performance for next build so hopefully you will see some improvements. Closing this for now, because main issue seems solved.
